ocamltest/filecompare.ml

```diff
@@ -108,25 +109,26 @@ let really_input_up_to ic =
     Bytes.sub buf 0 bytes_read
 
 let compare_binary_files bytes_to_ignore file1 file2 =
-  let ic1 = open_in_bin file1 in
-  let ic2 = open_in_bin file2 in
-  seek_in ic1 bytes_to_ignore;
-  seek_in ic2 bytes_to_ignore;
-  let rec compare () =
-    let block1 = really_input_up_to ic1 in
-    let block2 = really_input_up_to ic2 in
-    if block1 = block2 then
-      if Bytes.length block1 > 0 then
-        compare ()
-      else
-        Same
-    else
-      Different
-  in
-  let result = compare () in
-  close_in ic1;
-  close_in ic2;
-  result
+  Sys.with_input_file ~bin:true file1
+    (fun ic1 ->
+       Sys.with_input_file ~bin:true file2
+         (fun ic2 ->
```

Comment:
I think this would be more readable (to me!) with the

```
Sys.with_input_file ~bin:true file1 @@ fun ic1 ->
Sys.with_input_file ~bin:true file2 @@ fun ic2 ->
``` 

idiom. (Recently I've been using a `( let@ )` binding operator for this, but this would be too daring.)

(There are several other occurrences of your double-indentation-bump approach below in the diff; so if you were to agree, it would be nice to have a look at them too.)

Reply:
Nice! Amended as suggested. (And also added one more commit applying the same style to the `Sys.with_chdir` discussed above.)